atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitions.java

```diff
@@ -47,57 +48,23 @@ private ColumnFamilyDefinitions() {
      *
      *  Warning to developers: you must update CKVS.isMatchingCf if you update this method
      */
-    @SuppressWarnings("CyclomaticComplexity")
     static CfDef getCfDef(String keyspace, TableReference tableRef, int gcGraceSeconds, byte[] rawMetadata) {
-        Map<String, String> compressionOptions = Maps.newHashMap();
         CfDef cf = getStandardCfDef(keyspace, AbstractKeyValueService.internalTableName(tableRef));
 
-        boolean negativeLookups = false;
-        double falsePositiveChance = CassandraConstants.DEFAULT_LEVELED_COMPACTION_BLOOM_FILTER_FP_CHANCE;
-        int explicitCompressionBlockSizeKb = 0;
-        boolean appendHeavyAndReadLight = false;
-        TableMetadataPersistence.CachePriority cachePriority = TableMetadataPersistence.CachePriority.WARM;
-
-        if (!CassandraKeyValueServices.isEmptyOrInvalidMetadata(rawMetadata)) {
-            TableMetadata tableMetadata = TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(rawMetadata);
-            negativeLookups = tableMetadata.hasNegativeLookups();
-            explicitCompressionBlockSizeKb = tableMetadata.getExplicitCompressionBlockSizeKB();
-            appendHeavyAndReadLight = tableMetadata.isAppendHeavyAndReadLight();
-            cachePriority = tableMetadata.getCachePriority();
-        }
-
-        if (explicitCompressionBlockSizeKb != 0) {
-            compressionOptions.put(
-                    CassandraConstants.CFDEF_COMPRESSION_TYPE_KEY,
-                    CassandraConstants.DEFAULT_COMPRESSION_TYPE);
-            compressionOptions.put(
-                    CassandraConstants.CFDEF_COMPRESSION_CHUNK_LENGTH_KEY,
-                    Integer.toString(explicitCompressionBlockSizeKb));
-        } else {
-            // We don't really need compression here nor anticipate it will garner us any gains
-            // (which is why we're doing such a small chunk size), but this is how we can get "free" CRC checking.
-            compressionOptions.put(
-                    CassandraConstants.CFDEF_COMPRESSION_TYPE_KEY,
-                    CassandraConstants.DEFAULT_COMPRESSION_TYPE);
-            compressionOptions.put(
-                    CassandraConstants.CFDEF_COMPRESSION_CHUNK_LENGTH_KEY,
-                    Integer.toString(AtlasDbConstants.MINIMUM_COMPRESSION_BLOCK_SIZE_KB));
-        }
+        Optional<TableMetadata> tableMetadata = CassandraKeyValueServices.isEmptyOrInvalidMetadata(rawMetadata)
+                ? Optional.empty()
+                : Optional.of(TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(rawMetadata));
 
-        if (negativeLookups) {
-            falsePositiveChance = CassandraConstants.NEGATIVE_LOOKUPS_BLOOM_FILTER_FP_CHANCE;
-        }
+        Map<String, String> compressionOptions = getCompressionOptions(tableMetadata);
+        cf.setCompression_options(compressionOptions);
 
-        if (appendHeavyAndReadLight) {
+        if (tableMetadata.map(TableMetadata::isAppendHeavyAndReadLight).orElse(false)) {
```

Comment:
I'm somewhat biased to not inlining these, and having readable variable names.

Reply:
the same for all the `Optional.map(...).orElse(...)`

Reply:
OK, extracted each of the map/orElse expressions to a named variable. Makes sense